stdlib/test/builtin/test_list.mojo

```diff
@@ -39,7 +39,39 @@ fn test_repr_list() raises:
     assert_equal(empty.__repr__(), "[]")
 
 
+fn test_contains() raises:
+    # List
+    var p = List(2, 3, 5, 7)
+    assert_true(2 in p and 3 in p)
+    assert_true(p.__contains__(5) and p.__contains__(7))
+    assert_false(1 in p and 4 in p)
+    assert_false(p.__contains__(6) and p.__contains__(8))
+
+    var l = List[String]("Hello", ",", "World", "!")
```

Comment:
Should `List[String].__contains__` handle `StringLiteral` differently from `ListLiteral.__contains__`? Under the current implementation of `List.__contains__`, the following would hold:

```mojo
# Where `l` is defined as a List and `h` as a ListLiteral

assert_true("Hello" in l and String("Hello") in l)

assert_true("Mojo" not in h and String("Mojo") in h)
```

The behaviour is not exactly aligned between the two. Should the inherent nature of ListLiteral as a heterogeneous list result in a different __contains__ method compared to List[String]?

Reply:
Hi, thanks for this contribution !

I think to achieve that generically:
- on the side of ListLiteral: we'd need to somehow check if `Ts[i]` has a constructor that takes in `T` which I think isn't possible. 
- on the side of List: we'd need to "turn off" implicit casting of other types to the `EqualityComparableCollectionElement` which I also think is not possible currently.

Hacky solution ?: add a `StringableCollectionElement` constraint and use `str()` to compare everything. Kill performance, be hated by the next generation, but hey it works lol

just kidding, TLDR:  I think it's not achievable with current Mojo features :( , and I think accepting implicit casting is more pythonic

Reply:
Thanks for the reply! :) My concern was that the implementations between `List` and `ListLiteral` aren't directly aligned.  

I read through a few issues and it seems that strict type checking in `ListLiteral.__contains__` is somewhat necessary given the current state of Mojo.

I suppose this is a design choice moving forward: whether to keep strict comparisons in the heterogeneous-like `ListLiteral` or adopt a more Pythonic behavior like `List`. For the latter, I can leave a TODO?

Reply:
> I can leave a TODO?

great idea to avoid this popping up mysteriously as a bug later on, maybe even a `# FIXME` with the example you gave

Reply:
Okay, let's wait for @JoeLoser's feedback

Reply:
I think this is expected, although unfortunate.  Feel free to put a `FIXME` or `TODO` in the `ListLiteral` test case you mentioned.  Thanks for digging into this!